src/main/java/com/contrastsecurity/sdk/scan/CodeArtifactClient.java

```diff
@@ -43,12 +43,14 @@
    *
    * @param projectId ID of the project to which the code artifact belongs
    * @param file the file to upload
+   * @param metadata the prescan metadata to upload with the file artifact. Null may be given if
+   *     prescan data is not present.
    * @return new {@link CodeArtifactInner} from Contrast API
    * @throws IOException when an IO error occurs while making the request to the Contrast API
    * @throws UnauthorizedException when Contrast rejects the credentials used to send the request
    * @throws ResourceNotFoundException when the requested resource does not exist
    * @throws HttpResponseException when Contrast rejects this request with an error code
    * @throws ServerResponseException when Contrast API returns a response that cannot be understood
    */
-  CodeArtifactInner upload(String projectId, Path file) throws IOException;
+  CodeArtifactInner upload(String projectId, Path file, Path metadata) throws IOException;
```

Comment:
should this function signature been added as an additional one rather than replacing the original one for any kinda of backwards compat?

Reply:
is it valid use-case to upload a file without metadata?

Reply:
According to the Javadoc, it is valid to upload a file without metadata. Therefore, I think we should add a method override that introduces this parameter.